auth/store.go

```diff
@@ -807,20 +879,53 @@ func (as *authStore) isAuthEnabled() bool {
 	return as.enabled
 }
 
-func NewAuthStore(be backend.Backend) *authStore {
+func NewAuthStore(be backend.Backend, tokenType, jwtSignMethod, jwtPubKeyPath, jwtPrivKeyPath string) *authStore {
+	as := &authStore{
+		be:       be,
+		revision: 0,
+	}
+
+	switch tokenType {
+	case "simple":
+		as.tokenType = TokenSimple
+		as.simpleTokens = make(map[string]string)
+	case "jwt":
+		as.tokenType = TokenJwt
+		as.signMethod = jwtSignMethod
+
+		verifyBytes, err := ioutil.ReadFile(jwtPubKeyPath)
```

Comment:
can we do the configuration validation outside auth pkg? we probably should pass in the key/cert to this pkg.